### PR TITLE
perf: Auto-register bm.Scenario sub-classes

### PR DESCRIPTION
**Pull request by brettlangdon**
## Description
Auto-register every subclass of `bm.Scenario` so you don't need to sub-class _and_ decorate with `@bm.register`.

### CODE REVIEW

File: benchmarks/bm/_scenario.py [hunk 1]
```diff
@@ -6,18 +6,49 @@
...
 
+        # Make sure every sub-class is wrapped by `attr.s`
+        cls = attr.s()(cls)
 
-@attr.s
-class Scenario(six.with_metaclass(abc.ABCMeta)):
+        # Do not register the base Scenario class
+        # DEV: We cannot compare `cls` to `Scenario` since it doesn't exist yet
+        if cls.__module__ != __name__:
+            _register(cls)
```
Comment (P403n1x87) [inline benchmarks/bm/_scenario.py:48]: Can the check be moved inside `_register` so that `Scenario` exists?

File: benchmarks/bm/_scenario.py [hunk 2]
```diff
@@ -6,18 +6,49 @@
...
+        field.name: getattr(parsed_args, field.name)
+        for field in attr.fields(scenario_cls)
+        if hasattr(parsed_args, field.name)
+    }
+    scenario = scenario_cls(**config_dict)
+
+    runner.bench_time_func(scenario.scenario_name, scenario._pyperf)
+
+
+class ScenarioMeta(abc.ABCMeta):
```
Comment (P403n1x87) [inline benchmarks/bm/_scenario.py:38]: 👍 👍

File: benchmarks/bm/_scenario.py (hunk 1 above)
Comment (brettlangdon) [reply benchmarks/bm/_scenario.py:48]: it still won't exist until after this whole function exists, but we can move this `cls.__module__ != __name__` check to `_register`, or else we can do a name check like `cls.__name__ != "Scenario"`, but that seemed a little more error prone if someone did `class Scenario(bm.Scenario)`.

**Review by majorgreys — APPROVED** [review]
👍🏽

**Review by Yun-Kim — APPROVED** [review]

**Pull request merged**